### PR TITLE
Use the newly created TA codeowners

### LINKED ISSUE #1097: Setup codeowners/maintainers for operator components: auto-instrumentation, target allocator
The operator offers 3 main functionality at the moment:
1. collector deployment 
2. auto-instrumentation for Java, NodeJS, Python, Dotnet workloads
3. target-allocator for collecting Prometheus metrics

All these components require specialized knowledge (python instrumentation, java instrumentation, prometheus metrics collection) and a single maintainer usually does not have so wide knowledge and is not able to keep up with changes and fix bugs. 

I would propose to set up CODEOWNERS and assign individual contributors and/or official OTEL SIG groups to maintain parts of the operator:

```
autoinstrumentation/java                          @open-telemetry/java-maintainers @open-telemetry/java-instrumentation-maintainers
autoinstrumentation/python                      @open-telemetry/python-maintainers @open-telemetry/python-approvers
...
cmd/otel-allocator                                     @jaronoff97 @Aneurysm9 @secustor 
```

The intention is not to force SIGs (or ICs) to maintain parts of the operator, it should be rather voluntary. 

This approach is inspired from docs repository https://github.com/open-telemetry/opentelemetry.io/blob/main/.github/CODEOWNERS

### PR DESCRIPTION
**Pull request by jaronoff97**
This PR updates #1097 and edits the CODEOWNERS file to include ones for the target allocator

### CODE REVIEW

**Comment by pavolloffay** [comment]
The merge is blocked by https://github.com/open-telemetry/opentelemetry-operator/issues/1097. The group should be first created.

File: .github/CODEOWNERS
Comment (pavolloffay) [inline .github/CODEOWNERS:22]: @tigrannajaryan thanks for handling https://github.com/open-telemetry/community/issues/1218.

However this file is marked with errors:

Screenshot of Use the newly created TA codeowners by jaronoff97 · Pull Request #1145 · open-telemetry_opentelemetry-operator

It seems like the group @open-telemetry/operator-ta-maintainers should have write access to this repo.

Comment (tigrannajaryan) [reply .github/CODEOWNERS:22]: We give Write access to the entire repo only to Approvers. What you are asking for is outside of the current policies so I can't do it. Either find a different way or create an issue in the community repo to change the policies/make exception.

**Comment by jaronoff97** [comment]
this is blocked by this PR https://github.com/open-telemetry/community/issues/1260

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**
